src/main/java/seedu/address/model/person/PhoneNumberMatchesPredicate.java

```diff
@@ -0,0 +1,42 @@
+package seedu.address.model.person;
+
+import java.util.Objects;
+import java.util.function.Predicate;
+
+/**
+ * Tests that a {@code Person}'s {@code Phone} matches exactly the given phone number.
+ */
+public class PhoneNumberMatchesPredicate implements Predicate<Person> {
+    private final String phoneNumber;
+
+    public PhoneNumberMatchesPredicate(String phoneNumber) {
+        this.phoneNumber = phoneNumber;
+    }
+
+    @Override
+    public boolean test(Person person) {
+        return person.getPhone().value.equals(phoneNumber);
```

Comment:
Might be violating LOD

Maybe there is a need to add a `hasPhoneNumber(phoneNumber)` method in `Person`?